### PR TITLE
User story 27- Merchant Invoice Index

### PR DESCRIPTION
**Pull request by jaw772**
-Created views for both the index and show for invoices.
   -the show view is only the file name to make the test pass
-Created a merchant_invoices controller in case we need to have a bulk index page with all the invoices that have been created.

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -1,3 +1,9 @@
 class Item < ApplicationRecord
   belongs_to :merchant
+  has_many :invoice_items
+  has_many :invoices, through: :invoice_items
+  
+  def self.invoice_finder(merchant_id)
+    Invoice.joins(:invoice_items => :item).where(:items => {:merchant_id => merchant_id})
```
Comment (LelandCurtis) [inline app/models/item.rb:11]: would this work instead? `Merchant.find(merchant_id).invoices`

File: spec/features/merchants/invoices/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.describe 'Merchant Invoices Index page' do
+  invoice_item1 = FactoryBot.create(:invoice_item)
```
Comment (LelandCurtis) [inline spec/features/merchants/invoices/index_spec.rb:4]: don't need FactoryBot.create. Can just say `create`

File: spec/models/item_spec.rb [hunk 3]
```diff
@@ -2,4 +2,14 @@
 
 RSpec.describe Item, type: :model do
   pending "add some examples to (or delete) #{__FILE__}"
+
+  describe 'class methods' do
+    let!(:merchant){Merchant.create!(name: "Bill's Flowers")}
```
Comment (LelandCurtis) [inline spec/models/item_spec.rb]: can use factory bot: merchant = create(:merchant, name: "Bill's Flowers")

**Review by LelandCurtis — COMMENTED** [review]
- ideas for how to better leverage factoryBot. Minro items

File: app/models/item.rb (hunk 1 above)
Comment (EMielke76) [reply app/models/item.rb:11]: I don't think it will, Merchants don't have a relationship to invoices, so you can't call `invoices` on a `merchant`

**Review by LelandCurtis — APPROVED** [review]

**Pull request merged**